frontend/src/modules/announcements/components/AnnouncementAddForm.tsx

```diff
@@ -0,0 +1,221 @@
+import {
+  Button,
+  Flex,
+  FormControl,
+  FormLabel,
+  Spacer,
+  Textarea,
+} from '@chakra-ui/react';
+import { Field, FieldProps, Form, Formik } from 'formik';
+import { AnnouncementDTO, AnnouncementDTOViewAccessEnum } from 'generated-api';
+import { useEffect } from 'react';
+import * as Yup from 'yup';
+import {
+  Checkbox,
+  FormikResetEffect,
+  FormLayout,
+  Input,
+  Select,
+} from '../../../shared/components/form';
+import { FormikResetButton } from '../../../shared/components/form/FormikResetButton';
+import { useToast } from '../../../shared/hooks';
+import { useAnnouncements } from '../hooks';
+
+interface AnnouncementAddFormProps {
+  onComplete: (announcementDTO: AnnouncementDTO) => void;
+}
+
+export const AnnouncementAddForm = ({
+  onComplete,
+}: AnnouncementAddFormProps) => {
+  const { addAnnouncement } = useAnnouncements();
+  const toast = useToast();
+
+  const onSubmit = (announcementDTO: AnnouncementDTO) => {
+    addAnnouncement.mutate(announcementDTO);
+  };
+
+  const initialValues = {
+    title: '',
+    subtitle: '',
+    content: '',
+    tags: [],
+    viewAccess: AnnouncementDTOViewAccessEnum.Public,
+  };
+
+  const validationSchema = Yup.object({
+    title: Yup.string().min(1).max(40).required('Required'),
+    subtitle: Yup.string().min(1).max(40),
+    content: Yup.string().min(1).max(500).required('Required'),
+    tags: Yup.array()
+      .min(1, 'Must include at least one tag')
+      .required('Required'),
+    viewAccess: Yup.string().required('Required'),
+  });
+
+  /**
+   * To properly handle resetting of viewAccess enum
+   */
+  const handleReset = () => {
+    const selector = window.document.getElementById(
+      'viewAccess'
+    ) as HTMLSelectElement | null;
```

Comment:
this is fine but i think another way you can do it is by using the `useRef` hook to select the element